Typescript fails to compile when including unmock as a package dependency

<!-- If you have any questions while filling out this issue template, please feel free to reach out to us through the [Unmock Community Gitter](https://gitter.im/unmock/community) -->

### Description

Unmock fails to compile because of a type import 
<!-- Tell us about the bug you found -->

### Steps to Reproduce

- Include unmock as a package dependency
- Run `tsc`

<!-- Steps for how we can replicate your experience (numbered lists are best) -->

### Expected Result 

Should compile.

<!-- What did you expect to happen? -->

### Actual Result 

Fails with this error:

```bash
node_modules/unmock-node/dist/backend.d.ts:2:37 - error TS2307: Cannot find module 'unmock-core/src/interceptor'.
2 import { IInterceptorFactory } from "unmock-core/src/interceptor";
```

<!-- What actually happened? -->

### Additional Context

To fix the bug, I'd recommend hoisting all imports to a top-level namespace. As the `src` directory is not packaged (only the `dist` one is), `unmock-node` cannot find `unmock-core/src`.

<!-- 
    Anything else that will help us better understand, for example:
      * Information about your local environment
      * Screenshots
      * Code snippets
-->

--- a/packages/unmock-core/src/index.ts
+++ b/packages/unmock-core/src/index.ts
@@ -12,6 +12,7 @@ export * from "./types";
 export { sinon };
 export { u } from "./nock";
 export { transform, Addl, Arr } from "./generator-utils";
+export { IInterceptorFactory, IInterceptorOptions } from "./interceptor";
 export { IService } from "./service/interfaces";
 export { Service } from "./service";
 export { ServiceCore } from "./service/serviceCore";

--- a/packages/unmock-node/src/backend.ts
+++ b/packages/unmock-node/src/backend.ts
@@ -1,8 +1,4 @@
-import { Backend } from "unmock-core";
-import {
-  IInterceptorFactory,
-  IInterceptorOptions,
-} from "unmock-core/src/interceptor";
+import { Backend, IInterceptorFactory, IInterceptorOptions } from "unmock-core";
 import createFsServiceDefLoader from "./fs-service-def-loader";
 import NodeInterceptor from "./interceptor/node-interceptor";
 import FSLogger from "./loggers/filesystem-logger";

